eth/backend.go

```diff
@@ -84,17 +82,33 @@ func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) er
 	b.nonceManager.Unlock(sender)
 
 	data := tx.Data()
-	method, ok := b.methods[string(data[:4])]
-	if !ok {
-		method = "unknown"
+	methodSig := data[:4]
+	method, err := b.abiMap[string(methodSig)].MethodById(methodSig)
+	if err != nil {
+		return err
 	}
 
-	if err != nil {
-		glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\".  \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), method, err, strings.Repeat("*", 75))
+	var methodName string
```

Comment:
Looks like we can just use a short declaration for `methodName` here i.e. `methodName := method.Name`

Reply:
Actually if method is nil this will be a nil pointer so that's why I orignally added

```
	if !ok {		method, err := b.abiMap[string(methodSig)].MethodById(methodSig)
		method = "unknown"		if err != nil {
		return err
	}
```

but instead we could do
```if method != nil { methodName =  method.Name }

Reply:
Ah right - good point about not being able to find the ABI in `abiMap`.

In that case, we should have a check that confirms that we actually have a valid ABI. Something like:

```
var methodName string
var txParamsString string

methodSig := data[:4]
abi, ok := b.abiMap[string(methodSig)]
if !ok {
    methodName = "unknown"
} else {
    methodName = method.Name

    // Decode tx params and populate txParamsString
}

// Log statements
```